mlir/lib/Dialect/Affine/Transforms/AffineLoopInterchange.cpp

```diff
@@ -0,0 +1,804 @@
+                                int64_t &constantVectorValue,
+                                ArrayRef<Value> operands,
+                                DenseMap<Value, unsigned> &loopIndexMap) {
+  // TODO: Implement support for terminal symbols in `expr`.
+  switch (expr.getKind()) {
+  case AffineExprKind::Add: {
+    // Is this sub-expr a constant? If yes, no need to modify `row`. Start by
+    // assuming the sub-expr is not a constant.
+    bool isConstSubExpr = false;
+    AffineBinaryOpExpr addExpr = expr.cast<AffineBinaryOpExpr>();
+    AffineExpr lhs = addExpr.getLHS();
+    AffineExpr rhs = addExpr.getRHS();
+    unsigned lhsPosition = 0;
+    unsigned rhsPosition = 0;
+    
+    switch (lhs.getKind()) {
+    case AffineExprKind::DimId: {
+      auto dim = lhs.cast<AffineDimExpr>();
+      lhsPosition = loopIndexMap[operands[dim.getPosition()]];
+      break;
+    }
+    case AffineExprKind::Constant: {
+      constantVectorValue += rhs.cast<AffineConstantExpr>().getValue();
+      isConstSubExpr = true;
+    }
+    }
+    if (row[lhsPosition] == 0 && !isConstSubExpr)
+      row[lhsPosition] = 1;
+    
+    isConstSubExpr = false;
+    switch (rhs.getKind()) {
+    case AffineExprKind::DimId: {
+      auto dimExpr = rhs.cast<AffineDimExpr>();
+      rhsPosition = loopIndexMap[operands[dimExpr.getPosition()]];
+      break;
+    }
+    case AffineExprKind::Constant: {
+      // If the RHS is a constant, add this constant to constantVectorValue.
+      constantVectorValue += rhs.cast<AffineConstantExpr>().getValue();
+      isConstSubExpr = true;
+    }
+    }
+    if (row[rhsPosition] == 0 && !isConstSubExpr)
+      row[rhsPosition] = 1;
+    break;
+  }
+  case AffineExprKind::Mul: {
+    AffineBinaryOpExpr mulExpr = expr.cast<AffineBinaryOpExpr>();
+    AffineExpr lhs = mulExpr.getLHS();
+    AffineExpr rhs = mulExpr.getRHS();
+    unsigned dimIdPos = 0;
+    if (lhs.isa<AffineDimExpr>()) {
+      auto dim = lhs.cast<AffineDimExpr>();
+      dimIdPos = loopIndexMap[operands[dim.getPosition()]];
+    }
+    // RHS in this case should always be a constant.
+    if (rhs.isa<AffineConstantExpr>()) {
+      row[dimIdPos] = rhs.cast<AffineConstantExpr>().getValue();
+    }
+    break;
+  }
+  case AffineExprKind::DimId: {
+    auto dim = expr.cast<AffineDimExpr>();
+    row[loopIndexMap[operands[dim.getPosition()]]] = 1;
+    constantVectorValue += 0;
+    break;
+  }
+  case AffineExprKind::CeilDiv:
+  case AffineExprKind::FloorDiv:
+  case AffineExprKind::Mod: {
+    auto modExpr = expr.cast<AffineBinaryOpExpr>();
+    AffineExpr lhs = modExpr.getLHS();
+    AffineExpr rhs = modExpr.getRHS();
+    if (lhs.isa<AffineDimExpr>()) {
+      auto dim = lhs.cast<AffineDimExpr>();
+      row[loopIndexMap[operands[dim.getPosition()]]] = 1;
+    }
+    // RHS in this case is always a constant or a symbol. For a constant, we
+    // don't need to modify the access matrix.
+  }
```

Comment:
Add a default `unreachable` here. See other code for reference.

Reply:
Fixed. Thanks. Actually, this part is indeed reachable. But we don't need to process those cases. Removed the comment. Adding llvm_unreachable() would throw errors.